parcels/fieldset.py

```diff
@@ -210,6 +213,9 @@ def add_vector_field(self, vfield):
         :param vfield: :class:`parcels.field.VectorField` object to be added
         """
         setattr(self, vfield.name, vfield)
+        for v in vfield.__dict__.values():
+            if isinstance(v, Field) and (v not in self.get_fields()):
```

Comment:
what is this ?!
a) isn't that exactly what we have the get_fields()-method for ?
b) how could that if-statement even be true ?

Also, in this case the object-object comparison can be quite expensive. As far as I understand, once a Field is created, it's name is fixed, so why not ?
```
ref_names = [f.name for f in self.get_fields()]
for v in vfield.get_fields():  # if that function doesn't exist, it seams plausible to add it
    if v.name not in ref_names:
        self.add_field(v)
```

Then you (a) don't forcefully treat `vfield` as a dict (which may run its own errors) and (b) avoid complex object-in-object comparisons.

Reply:
I've now changed the code as you suggested. Do note that the `VectorField.get_fields()` still requires a call to `__dict__`; I think we can assume that users only call this function with Fields; and otherwise get an error